main.tf

```
@@ -38,6 +38,24 @@ resource "google_storage_bucket" "gcs_bucket" {
       cors             # ignore CORS changes. Use 'gsutil' tool instead. See https://cloud.google.com/storage/docs/configuring-cors
     ]
   }
+  dynamic "lifecycle_rule" {
+    for_each = var.lifecycle_rules
+    content {
+      action {
+        type          = lifecycle_rule.value.action.type
+        storage_class = lookup(lifecycle_rule.value.action, "storage_class", null)
+      }
+      condition {
+        age            = lookup(lifecycle_rule.value.condition, "age", null)
+        created_before = lookup(lifecycle_rule.value.condition, "created_before", null)
+        with_state     = lookup(lifecycle_rule.value.condition, "with_state", null)
+        matches_storage_class = contains(keys(lifecycle_rule.value.condition), "matches_storage_class") ? split(
+          ",", lifecycle_rule.value.condition["matches_storage_class"]
+        ) : null
```

```suggestion
        matches_storage_class = contains(keys(lifecycle_rule.value.condition), "matches_storage_class") ? (
          split(",", lifecycle_rule.value.condition["matches_storage_class"])
        ) : null
```
to read the full `split()` command in a single line